fastapi/encoders.py

```diff
@@ -105,24 +105,22 @@ def jsonable_encoder(
                 )
             )
         return encoded_list
+    for encoder_type, encoder in custom_encoder.items():
+        if isinstance(obj, encoder_type):
+            return encoder(obj)
+    for encoder_type, encoder in ENCODERS_BY_TYPE.items():
+        if isinstance(obj, encoder_type):
+            return encoder(obj)
     errors: List[Exception] = []
     try:
-        if custom_encoder and type(obj) in custom_encoder:
-            encoder = custom_encoder[type(obj)]
-        else:
-            encoder = ENCODERS_BY_TYPE[type(obj)]
-        return encoder(obj)
-    except KeyError as e:
+        data = dict(obj)
+    except Exception as e:
         errors.append(e)
```

Comment:
Should this be here? It looks like you'll add the exception even if the next block succeeds?

Reply:
Well as soon as one of the attempts to fill the `data` variable succeeds the function returns and this list of errors is garbage collected. This is the way the originial implementation worked and I think it's an ok implementation?